lib/Serialization/Deserialization.cpp

```diff
@@ -5174,6 +5182,17 @@ class TypeDeserializer {
     if (!diffKind.hasValue())
       MF.fatal();
 
+    const clang::FunctionType *clangFunctionType = nullptr;
+    if (clangFunctionTypeID) {
+      auto clangType = MF.getClangType(clangFunctionTypeID);
+      if (!clangType)
+        return clangType.takeError();
+      clangFunctionType =
+        dyn_cast_or_null<clang::FunctionType>(clangType.get());
```

Comment:
This seems insufficient in the presence of block pointers. Is that something we are not going to do in the future? [I think we discussed this and the answer was no, but double-checking in case I am misremembering. If we are not doing that, I should update a bunch of code/comments.]

Reply:
We should absolutely still do block pointers.  I didn't think about tthem when I was writing this because the existing `clangFunctionType` variable was a `const clang::FunctionType *`, apparently because `SILFunctionType::ExtInfo` requires that.  So I actually can't generalize this given the current interfaces; since you have an outstanding patch in that area, I'll let you handle it.